app/src/main/java/org/fossasia/openevent/general/event/EventUtils.kt

```diff
@@ -63,13 +73,69 @@ object EventUtils {
         }
     }
 
+    fun getFormattedDateWithoutYear(date: ZonedDateTime): String {
+        val dateFormat: DateTimeFormatter = DateTimeFormatter.ofPattern("EEEE, MMM d")
+        try {
+            return dateFormat.format(date)
+        } catch (e: IllegalArgumentException) {
+            Timber.e(e, "Error formatting Date")
+            return ""
+        }
+    }
+
     fun getFormattedTime(date: ZonedDateTime): String {
-        val timeFormat: DateTimeFormatter = DateTimeFormatter.ofPattern("hh:mm a Z")
+        val timeFormat: DateTimeFormatter = DateTimeFormatter.ofPattern("hh:mm a")
         try {
             return timeFormat.format(date)
         } catch (e: IllegalArgumentException) {
             Timber.e(e, "Error formatting time")
             return ""
         }
     }
+
+    fun getFormattedTimeZone(date: ZonedDateTime): String {
+        val timeFormat: DateTimeFormatter = DateTimeFormatter.ofPattern("z")
+        try {
+            return timeFormat.format(date)
+        } catch (e: IllegalArgumentException) {
+            Timber.e(e, "Error formatting time")
+            return ""
+        }
+    }
+
+    fun getFormattedEventDateTimeRange(startsAt: ZonedDateTime, endsAt: ZonedDateTime): String {
+        try {
+            if (EventUtils.getFormattedDate(startsAt) != EventUtils.getFormattedDate(endsAt))
+                return "${getFormattedDateShort(startsAt)}, ${getFormattedTime(startsAt)}"
+            else
+                return "${getFormattedDateWithoutYear(startsAt)}"
+        } catch (e: IllegalArgumentException) {
+            Timber.e(e, "Error formatting time")
+            return ""
+        }
+    }
+
+    fun getFormattedEventDateTimeRangeSecond(startsAt: ZonedDateTime, endsAt: ZonedDateTime): String {
+        try {
+            if (EventUtils.getFormattedDate(startsAt) != EventUtils.getFormattedDate(endsAt))
+                return "- ${getFormattedDateShort(endsAt)}, ${getFormattedTime(endsAt)} ${getFormattedTimeZone(endsAt)}"
+            else
+                return "${getFormattedTime(startsAt)} - ${getFormattedTime(endsAt)} ${getFormattedTimeZone(endsAt)}"
+        } catch (e: IllegalArgumentException) {
+            Timber.e(e, "Error formatting time")
+            return ""
+        }
+    }
+
+    fun getFormattedDateTimeRangeDetailed(startsAt: ZonedDateTime, endsAt: ZonedDateTime): String {
+        try {
+            if (EventUtils.getFormattedDate(startsAt) != EventUtils.getFormattedDate(endsAt))
+                return "${getFormattedDate(startsAt)} at ${getFormattedTime(startsAt)} - ${getFormattedDate(endsAt)} at ${getFormattedTime(endsAt)} (${getFormattedTimeZone(endsAt)})"
+            else
+                return "${getFormattedDate(startsAt)} from ${getFormattedTime(startsAt)} to ${getFormattedTime(endsAt)} (${getFormattedTimeZone(endsAt)})"
+        } catch (e: IllegalArgumentException) {
```

Comment:
This is at least calculating the date twice

Store the reference to be reused
Similarly for all mehods